packages/SwingSet/src/vats/comms/state.js

```diff
@@ -83,10 +87,16 @@ export function makeState(syscall, identifierBase = 0) {
   // p.nextID = $NN  // kernel-facing promise identifier allocation counter (p+NN)
   // c.$kfref = $lref // inbound kernel-facing c-list (o+NN/o-NN/p+NN/p-NN -> loNN/lpNN)
   // c.$lref = $kfref // outbound kernel-facing c-list (loNN/lpNN -> o+NN/o-NN/p+NN/p-NN)
+  // cr.$lref = 1 | <missing> // isReachable flag
```

Comment:
In the kernel the reachable flag is encoded in the value of the clist entry, whereas here it appears to be a separate key.  Might be worth a comment talking about the change in design pattern.

Reply:
Yeah, I'm experimenting with a different approach here.. I think I like the separate flag better (easier to manipulate), although it's probably less space-efficient on the DB.

There are a few other differences in approach that I think I'll backport to the kernel implementation eventually. Specifically the way an downstream-travelling `retireImport` simply deletes the upstream-facing c-list entry and marks the ref as `maybeFree`, then lets the main processing loop discover that the exporter's c-list is empty but the recognizable count is still nonzero (which reacts by deleting all the importer's c-list entries and emitting `retireExports`). In the kernel, `disptach.retireImport` deletes the importer's c-list entries immediately. Having written it both ways, I think I like the comms approach better.